frontend/smartComponents/SmartGameBox.tsx

```diff
@@ -20,15 +19,14 @@ export class SmartGameBox extends React.Component<SmartGameBoxProps, SmartGameBo
 			gameBoxProps: null,
 		};
 	}
-	public componentDidMount() {
-		fetchGameBoxData(this.props.gameId, this.props.imageId)
-			.then(gameBoxProps => {
-				this.setState({ gameBoxProps, isDataAvailable: true });
-			})
-			.catch(error => {
-				console.log(error);
-				this.setState({ isDataAvailable: false });
-			});
+	public async componentDidMount() {
+		try {
+			const gameBoxProps = await fetchGameBoxData(this.props.gameId);
```

Comment:
You need to create pretty complex props here. If we're going the route of each component requesting its own data don't you think it'd make more sense to break `GameBox` into smaller components inside, each component fetching the data it requires?